### PR TITLE
Enhancements to Psi4 Handling, Thread Management, and Alignment for Calculations

### PR DESCRIPTION
**Pull request by Marclie**
This pull request includes three significant improvements for the pysisyphus package:

1.    The `no_reorient` and `no_com` options have been added to the generated input files from the Psi4 calculator, preventing undesired modification of geometry data.
2.    The thread distribution strategy has been revised when using the Dask scheduler with the `pal` flag, improving performance and efficiency for concurrent calculations.
3.    A new flag, `align_factor`, has been implemented to allow partial alignment in calculations that have an energy dependence on the molecular orientation, addressing a key issue in optimization convergence for such cases.

Please refer to the individual commit messages for more detailed explanations of each change. Thank you for considering these enhancements.

### CODE REVIEW

File: pysisyphus/cos/ChainOfStates.py [hunk 1]
```diff
@@ -231,6 +231,38 @@ def set_images(self, indices, images):
...
+            # divide pal of each image by the number of workers available or available images
+            n_workers = len(client.scheduler_info()["workers"]) # number of workers available
+            n_images = len(images_to_calculate) # number of images to calculate
+
+            #split images to calculate into batches of n_workers and set pal of each image
+            n_batches = n_images // n_workers
+            for i in range(0, n_batches):
+                for j in range(i*n_workers, (i+1)*n_workers):
+                    new_pal = images_to_calculate[j].calculator.pal // n_workers # divide pal by the number of workers
+                    images_to_calculate[j].calculator.pal = new_pal if new_pal > 0 else 1 # set pal to 1 if it is smaller than 1
```
Comment (eljost) [inline pysisyphus/cos/ChainOfStates.py:250]: `new_pal` should be a constant in the scope of the function, shouldn't it? All images will/should have the same `pal` value ...
If so `new_pal` should be calculated outside of the loop
`new_pal = min(1, images_to_calculate[0].calculator.pal // n_workers)`

File: pysisyphus/cos/ChainOfStates.py [hunk 2]
```diff
@@ -231,6 +231,38 @@ def set_images(self, indices, images):
...
+
+            # distribute the remaining image pals to the last batch size
+            n_last_batch = n_images % n_workers
+            if n_last_batch > 0:
+                for i in range(n_batches*n_workers, n_images):
+                    new_pal = images_to_calculate[i].calculator.pal // n_last_batch # divide pal by the last batch size
```
Comment (eljost) [inline pysisyphus/cos/ChainOfStates.py:256]: A similar argument applies here.

**Comment by Marclie** [comment]
Hello Johannes,

Thank you for your review and constructive feedback.

The alignment factor is definitely a niche use-case specific to my research, where I am generating reaction pathways of light-catalyzed reactions using quantum electrodynamics. In these scenarios, light is bound to the molecular dipole moment and the dipole's alignment with the light's polarization affects the total energy. Forced full alignment causes instability when the gradients should rotate the molecule. However, I still needed some alignment between images. The `align_factor` solved the problem for me after some fine-tuning.

As I am still figuring out the details of Pysisyphus, I chose to use a static variable for `align_factor` to avoid possibly breaking any functionalities. I appreciate your insight to know that only a few classes needed small changes. Similarly, I wasn't certain whether `pal` is consistent for all images in the COS calculations (the ONIOM model gave me a few doubts, as mentioned in my commit). It is much cleaner treating each image equally when distributing threads for Dask.

I hope these changes are useful and your time and attention is much appreciated.

Best Regards,
Marcus

**Comment by eljost** [comment]
Dear Marcus,
i sent you some additional commits in a PR (https://github.com/Marclie/pysisyphus/pull/1). When you merge them, I can merge this PR into pysisyphus.

**Comment by Marclie** [comment]
Hello Johannes,
I have merged your fixes to my PR. As I gave in a comment, there is just one extra small change that I made that I think needs to be included in this PR.
The optimizer classes now do a great job at handling my orientation-dependent calculations. However, I discovered that the fixed images in an NEB could still be reoriented in the `set_coords_at` function from `ChainofStates.py` when aligning along the path. I've been using free-end neb, so this effect went unnoticed. The small change in the commit before I merged addresses this.

**Comment by eljost** [comment]
Thanks for your contributions!

**Pull request merged**